modules/View.js

```diff
@@ -368,6 +394,7 @@
      * @method activate
      */
     activate: function() {
+      this.trigger('before-activate');
```

Comment:
If we really want to trigger events right before and after lifecycle hooks, would it be a good idea to wrap the hooks with a function that automatically generates/triggers the events, similar to Java AOP? Advantage: less duplication, less chance of forgetting to add an event. Disadvantage: harder to understand.